MoreLinq/ToReadOnlyCollection.cs

```diff
@@ -0,0 +1,21 @@
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///     Converts an IEnumerable to a ReadOnlyCollection.
+        /// </summary>
+        /// <typeparam name="T">Type of the items in the source enumeration.</typeparam>
+        /// <param name="source">The IEnumerable nullable collection of T to be converted.</param>
+        /// <returns>A ReadOnlyCollection of T as the source.</returns>
+        public static ReadOnlyCollection<T> ToReadOnlyCollection<T>(this IEnumerable<T> source) =>
+            source is null
+                ? throw new ArgumentNullException(nameof(source))
+                : new ReadOnlyCollection<T>(source as IList<T> ?? source.ToList());
```

Comment:
This operator should not keep a reference to the original `source` after the method ends. On the off chance the consumer provides a list, they may modify the list after they provide it, which would violate the intent of the `ReadOnlyCollection`. 

Because of this ambiguity, I would recommend not implementing this operator in a library at all - the semantics of the use of `source` are not obvious and not everyone will expect the same behavior.

Reply:
good point
im changing the implementation to prevent that original source reference leak.

would that change your recommandation? I find myself using ReadOnlyCollection alot, and i would love to be able to use it via MoreLinq

Reply:
Not really, for me at least. The ambiguity is there regardless. In any case, remove it from this PR and create a new PR for a more specific conversation.

Reply:
Done